### PR TITLE
Redo manage appliance

### PR DESCRIPTION
**Pull request by keirthana**
In the process of redoing the [Manage appliance](https://anbox-cloud.io/docs/howto/manage/manage-appliance) topic, I have created the command reference for `anbox-cloud-appliance` as well. There was no way to do one without the other exercise as the existing howto topic contained reference information about the commands used in the appliance.

Some points to consider:

- Some of the CLI help text is not well-written and needs to be fixed for better clarity and use.
- Few commands need more clarity on the options and values passed with them. I am adding in-line comments in PR to point such commands.

### CODE REVIEW

File: ref/appliance-command-reference/cluster.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    anbox-cloud-appliance cluster <subcommand>
+
+## Subcommands
+
+### `add`
+Adds a new member to the cluster
+
+#### Usage
+
+    anbox-cloud-appliance cluster add <member name>
```
Comment (keirthana) [inline ref/appliance-command-reference/cluster.md:16]: As a user, how can I get the member name? Is this something I should know?

File: ref/appliance-command-reference/init.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    anbox-cloud-appliance init <options>
+
+## Options
+
+|Option | Description |
+|-------|-------------|
+|`--auto`|To enable automatic (non-interactive) mode. Use this option if you want to accept the default options for the appliance.|
+|`--charm-channel <channel>`|To specify the store channel from which Anbox Cloud charms are to be deployed|
+|`--force-network-bridge`|To force using a local network bridge instead of a FAN overlay network|
+|`--image-server-auth <credentials>`|To specify authentication credentials for the Anbox Cloud image server|
```
Comment (keirthana) [inline ref/appliance-command-reference/init.md:16]: Is there a format to specify the credentials?

File: ref/appliance-command-reference/cluster.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    anbox-cloud-appliance cluster <subcommand>
+
+## Subcommands
+
+### `add`
+Adds a new member to the cluster
+
+#### Usage
+
+    anbox-cloud-appliance cluster add <member-name>
```
Comment (keirthana) [inline ref/appliance-command-reference/cluster.md:16]: As a user, how can I get the member name? Is this something I should know?

File: ref/appliance-command-reference/ams.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+The `ams` command provides access to the Anbox Management Service (AMS) through subcommands. It exposes the AMS HTTPS service on the public endpoint of the machine on which the appliance is running. Using this command, you can [control AMS remotely](https://discourse.ubuntu.com/t/managing-ams-access/17774).
```
Comment (adglkh) [inline ref/appliance-command-reference/ams.md:1]: , you can [control AMS remotely](https://discourse.ubuntu.com/t/managing-ams-access/17774).
->
, you can expose or unexpose an AMS service through the load balancer. 

The above [link](remote control to AMS) has nothing to do with service exposure

File: ref/appliance-command-reference/cluster.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
+The `cluster` command allows management of manage cluster members for the Anbox Cloud Appliance. Using this command, you can join additional nodes to an existing installation or to remove existing nodes.
```
Comment (adglkh) [inline ref/appliance-command-reference/cluster.md:1]: to remove existing nodes.
->
to remove existing nodes from a cluster.

File: ref/appliance-command-reference/cluster.md (hunk 3 above)
Comment (adglkh) [reply ref/appliance-command-reference/cluster.md:16]: You need to choose one for yourselves, taking into account the current number of LXD nodes in the cluster, and name the new one starting from `lxd<existing_number_of_lxd_nodes>.`
https://discourse.ubuntu.com/t/how-to-join-a-machine-to-the-anbox-cloud-appliance/29054

File: ref/appliance-command-reference/ams.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+## Usage
+
+    anbox-cloud-appliance ams <subcommand>
+
+## Subcommands
+
+### `expose`
+Enables access to AMS in the load balancer.
+
+#### Usage
```
Comment (adglkh) [inline ref/appliance-command-reference/ams.md:12]: hmm, do we really need four Heading levels in this simple page?

File: ref/appliance-command-reference/init.md (hunk 2 above)
Comment (adglkh) [reply ref/appliance-command-reference/init.md:16]: No, there isn't. Technically, the image server auth token has no format. Usually, it's just a basic auth which consists of `<user>:<password>`

File: ref/appliance-command-reference/init.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+|`--image-server-url <url>`| To specify the URL of the Anbox Cloud image server|
```
Comment (adglkh) [inline ref/appliance-command-reference/init.md:17]: Also I am 100% sure if we want to list all options here for some of them like `--image-server-ath` and `--image-server-url`, they're mainly used by us/CI for testing

File: ref/appliance-command-reference/monitor.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
+The `monitor` command allows managing the monitoring dashboard. You can expose or unexpose the monitoring dashboard using the subcommands. By default, the monitoring dashboard is exposed.
+
+## Usage
+
+    anbox-cloud-appliance monitor <subcommand>
```
Comment (adglkh) [inline ref/appliance-command-reference/monitor.md:5]: @morphis , correct me if I was wrong, we've dropped the monitor command for the upcoming minor release(1.20.0), if so, we're going to drop this once 1.20.0 is out?

**Review by adglkh — CHANGES_REQUESTED** [review]

File: ref/appliance-command-reference/ams.md (hunk 6 above)
Comment (keirthana) [reply ref/appliance-command-reference/ams.md:12]: I felt that creating separate pages for subcommands would be overkill. So for being consistent across all topics, I have used the same heading format in all command reference topics.

File: ref/appliance-command-reference/init.md (hunk 7 above)
Comment (keirthana) [reply ref/appliance-command-reference/init.md:17]: If it is publicly available for the user to see and use, we should list it. Else, we can drop it. WDYT?

File: ref/appliance-command-reference/monitor.md (hunk 8 above)
Comment (morphis) [reply ref/appliance-command-reference/monitor.md:5]: Lets keep it until 1.20 is out and then we can remove it.

File: ref/appliance-command-reference/init.md (hunk 7 above)
Comment (morphis) [reply ref/appliance-command-reference/init.md:17]: We should document them if they show up in `--help`

File: ref/appliance-command-reference/cluster.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
+The `cluster` command allows management of manage cluster members for the Anbox Cloud Appliance. Using this command, you can join additional nodes to an existing installation or to remove existing nodes from a cluster.
+
+ See [How to join a machine to the Anbox Cloud Appliance](https://discourse.ubuntu.com/t/29054) for more information.
+
+ ## Usage
```
Comment (ajanon) [inline ref/appliance-command-reference/cluster.md:5]:
```suggestion
## Usage
```

File: ref/appliance-command-reference/cluster.md [hunk 10]
```diff
@@ -0,0 +1,27 @@
+The `cluster` command allows management of manage cluster members for the Anbox Cloud Appliance. Using this command, you can join additional nodes to an existing installation or to remove existing nodes from a cluster.
+
+ See [How to join a machine to the Anbox Cloud Appliance](https://discourse.ubuntu.com/t/29054) for more information.
```
Comment (ajanon) [inline ref/appliance-command-reference/cluster.md:3]:
```suggestion
See [How to join a machine to the Anbox Cloud Appliance](https://discourse.ubuntu.com/t/29054) for more information.
```

File: ref/appliance-command-reference/ams.md (hunk 6 above)
Comment (ajanon) [reply ref/appliance-command-reference/ams.md:12]: I kinda agree with Gary here: do we need that `Usage` heading? I feel like we could remove it everywhere.

File: ref/appliance-command-reference/destroy.md [hunk 11]
```diff
@@ -0,0 +1,11 @@
+The `destroy` command deletes the the deployment that the Anbox Cloud Appliance has created on a machine. If you want to uninstall the Anbox Cloud Appliance, you must first run this command before you uninstall the snap.
```
Comment (ajanon) [inline ref/appliance-command-reference/destroy.md:1]:
```suggestion
The `destroy` command deletes the deployment that the Anbox Cloud Appliance has created on a machine. If you want to uninstall the Anbox Cloud Appliance, you must first run this command before you uninstall the snap.
```

File: ref/appliance-command-reference/cluster.md [hunk 12]
```diff
@@ -0,0 +1,27 @@
+The `cluster` command allows management of manage cluster members for the Anbox Cloud Appliance. Using this command, you can join additional nodes to an existing installation or to remove existing nodes from a cluster.
```
Comment (ajanon) [inline ref/appliance-command-reference/cluster.md:1]: > The `cluster` command allows management of manage cluster members for the Anbox Cloud Appliance.

I think "manage" is a leftover from a previous version the sentence.

File: ref/appliance-command-reference/gateway.md [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+
+### `unexpose`
+Disables access to the stream gateway in the load balancer.
+
+#### Usage
+
+    anbox-cloud-appliance gateway unexpose
+
+
+
```
Comment (ajanon) [inline ref/appliance-command-reference/gateway.md:49]: Superfluous empty lines

Comment (ajanon) [inline ref/appliance-command-reference/gateway.md:49]: I'm not sure how we could do this better, but looking at the [rendered file](https://github.com/anbox-cloud/discourse-docs/blob/04aaea81fc3b50d60d386ec2b005c0cf72303d32/ref/appliance-command-reference/gateway.md), I think it's a bit hard to understand at what level a heading is and to what it applies while just reading through it.

It may render differently on our docs website, though.

File: ref/appliance-command-reference/help.md [hunk 14]
```diff
@@ -0,0 +1,7 @@
+The `help` command displays detailed information about the `anbox-cloud-appliance` tool.
+
+## Usage
+
+    anbox-cloud-appliance help
+
+You can also use the `--h` or `--help` option on any command to read its command line help.
```
Comment (ajanon) [inline ref/appliance-command-reference/help.md:7]:
```suggestion
You can also use the `-h` or `--help` option on any command to read its command line help.
```

File: ref/appliance-command-reference/init.md (hunk 7 above)
Comment (ajanon) [reply ref/appliance-command-reference/init.md:17]: Maybe we should have an "advanced options" section then? Or maybe add "(Advanced)" or something similar in the description.

That way, users know which options to look at first, before trying random options if things don't work the way they want.

File: ref/appliance-command-reference/landing.md [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+
+| Command | Description|
+|---------|------------|
+|[`ams`](tbd)    | Manage access to the Anbox Management Service (AMS)|
+|[`cluster`](tbd)| Manage cluster nodes|
+|[`dashboard`](tbd)| Manage the web dashboard of appliance|
+|[`destroy`](tbd) | Destroy the appliance|
+|[`gateway`](tbd) | Manage the appliance stream gateway|
+|[`help`](tbd) | Help about the `anbox-cloud-appliance` tool|
+|[`init](tbd)| Initialise the appliance|
```
Comment (ajanon) [inline ref/appliance-command-reference/landing.md:15]:
```suggestion
|[`init`](tbd)| Initialise the appliance|
```

File: ref/appliance-command-reference/monitor.md [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+
+    anbox-cloud-appliance monitor account create <account-name> [--password=<value>] [--email=<email>] [--role=viewer|admin|editor] [options]
+
+###### Options
+
+|Option | Description |
+|-------|-------------|
+| `-e`, `--email <email>`|Email address of the monitor account|
+| `-p`, `--password <value>`|Password of the monitor account|
+| `-r`, `--role <value>`|Role of the monitor account can be either of these three options: viewer/admin/editor. The default role is viewer.|
```
Comment (ajanon) [inline ref/appliance-command-reference/monitor.md:31]: Do we explain these roles somewhere? If we do, it may be useful to add a link, WDYT?

**Review by ajanon — CHANGES_REQUESTED** [review]

File: ref/appliance-command-reference/ams.md (hunk 6 above)
Comment (keirthana) [reply ref/appliance-command-reference/ams.md]: I have removed the `Usage` heading everywhere, does it read better now?

File: ref/appliance-command-reference/init.md (hunk 7 above)
Comment (keirthana) [reply ref/appliance-command-reference/init.md:15]: What are the advanced options for the commands that we have? AFAICS, only the `init` has a lot of options. Others have 1-3 options at max.

File: ref/appliance-command-reference/gateway.md (hunk 13 above)
Comment (keirthana) [reply ref/appliance-command-reference/gateway.md]: Does it read better after removing the `Usage` heading? If not, perhaps, I can turn the subcommands section into a table format.

File: ref/appliance-command-reference/init.md (hunk 7 above)
Comment (ajanon) [reply ref/appliance-command-reference/init.md:15]: Sorry, let me rephrase: `--image-server-path`, `--image-server-url`, probably also `--charm-channel` are options which we mostly use for development purposes. Even if they are publicly available, most, if not all, users should never need them. It may be better in that case to be very explicit that these are not useful for regular use cases.

File: ref/appliance-command-reference/gateway.md [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+    anbox-cloud-appliance gateway account create <account-name>
+
+This command creates an account with the given <account-name> and returns a token which you can use for accessing the Anbox Stream Gateway HTTP API directly or through the Anbox Stream SDK. See [How to access the stream gateway](https://discourse.ubuntu.com/t/managing-stream-gateway-access/17784) for more information.
+
+##### `delete`
+Deletes the specified account and removes access to the Anbox stream gateway for the specified account.
+
+    anbox-cloud-appliance gateway account delete <account-name>
+
```
Comment (ajanon) [inline ref/appliance-command-reference/gateway.md:25]: Do you think it would make sense to combine the subcommands?
Instead of having the `account` command, with two subcommands, can't we show that as two different commands: `account create` and `account delete`? It's not technically how it's implemented.

```suggestion
### `account create`
Creates an account to access the Anbox stream gateway.

    anbox-cloud-appliance gateway account create <account-name>

This command creates an account with the given <account-name> and returns a token which you can use for accessing the Anbox Stream Gateway HTTP API directly or through the Anbox Stream SDK. See [How to access the stream gateway](https://discourse.ubuntu.com/t/managing-stream-gateway-access/17784) for more information.

### `account delete`
Deletes the specified account and removes access to the Anbox stream gateway for the specified account.

    anbox-cloud-appliance gateway account delete <account-name>

```

This removes two heading levels and keep the important information. What do you think?

File: ref/appliance-command-reference/monitor.md [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+
+##### `delete`
+Deletes the specified account and removes access to the monitoring dashboard for the mentioned account.
+
+    anbox-cloud-appliance monitor account delete <account-name>
+
```
Comment (ajanon) [inline ref/appliance-command-reference/monitor.md:31]: Same question as above: what do you think about combining these subcommands into two commands: `account create` and `account delete`?

**Review by ajanon — CHANGES_REQUESTED** [review]

File: ref/appliance-command-reference/gateway.md (hunk 13 above)
Comment (ajanon) [reply ref/appliance-command-reference/gateway.md]: I think it reads much better, thanks!

File: ref/appliance-command-reference/ams.md (hunk 6 above)
Comment (ajanon) [reply ref/appliance-command-reference/ams.md]: I think it reads much better, thanks!

File: ref/appliance-command-reference/gateway.md (hunk 17 above)
Comment (keirthana) [reply ref/appliance-command-reference/gateway.md]: How does it read now? I kind of did what you have suggested but not entirely. I think it would be useful to retain the idea that the `create` and `delete` are subcommands to help the readers understand. At the same time, I do see your point about too many levels. Let me know what you think about the latest update and if it's good, I will implement the same everywhere there is a third level of subcommands.

Comment (ajanon) [reply ref/appliance-command-reference/gateway.md]: This is great, thanks!

File: ref/appliance-command-reference/monitor.md (hunk 16 above)
Comment (keirthana) [reply ref/appliance-command-reference/monitor.md:25]: We haven't explained it elsewhere as far as I could check, we could add it if there's more differentiation than the obvious to the roles. @morphis Is there a role definition here that we need to explain?

Comment (morphis) [reply ref/appliance-command-reference/monitor.md:24]: Given that we drop these soon I don't see a strong need for that.

**Review by morphis — APPROVED** [review]
LGTM

**Review by adglkh — APPROVED** [review]
LGTM
